components/Modals/WelcomeModal/WelcomeModal.tsx

```diff
@@ -20,7 +20,7 @@ const WelcomeModal: React.FC = () => {
 
   useEffect(() => {
     const hasChosenLanguages =
-      data?.currentUser?.languagesNative.length && data?.currentUser?.languagesLearning.length
+      data?.currentUser?.languages.length 
```

Comment:
We need to make sure there's at least one native and one learning language here. A sorta tricky way to do it:

```suggestion
      (data?.currentUser?.languages.length || [])
        .reduce((acc, { level }) => acc | (level === LanguageLevel.Native ? 1 : 2), 0) === 3
```

Reply:
Yeah I agree, I think we could discuss how the end UX/flow should be and how we can wire that up to this